### PR TITLE
remove duplicate function in scala side

### PR DESCRIPTION
**Pull request by zhexuany**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/TiStrategy.scala [hunk 1]
```diff
@@ -200,43 +200,29 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
...
-    val tiColumnSet: Seq[TiExpression] = attributeSet.toSeq.collect {
+    val tiColumnSeq: Seq[TiExpression] = attributeSet.toSeq.collect {
       case BasicExpression(expr) => expr
     }
     val resolver = new MetaResolver(source.table)
-    val tiColumns: Seq[TiColumnRef] = extractTiColumnRefFromExpressions(tiColumnSet)
+    var tiColumns: mutable.HashSet[TiColumnRef] = mutable.HashSet.empty[TiColumnRef]
+    for (expression <- tiColumnSeq) {
+      val colSetPerExpr = PredicateUtils
+        .extractColumnRefFromExpression(expression.asInstanceOf[com.pingcap.tikv.expression.Expression])
```
Comment (birdstorm) [inline core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: `.asInstanceOf[com.pingcap.tikv.expression.Expression]` is useless

Comment (zhexuany) [reply core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: DONE.

**Pull request merged**